src/alchemlyb/parsing/util.py

```diff
@@ -47,7 +47,10 @@ def anyopen(filename, mode='r'):
     ext = os.path.splitext(filename)[1]
 
     if ext in extensions:
-       opener= extensions[ext]
```

Comment:
This appears to have been a bug, right?

I'd prefer this to be a separate PR so that each PR addresses one specific problem and not a hodgepodge of things. Open an issue that identifies the problem and submit a PR + test for this issue. (Remove it from *this* PR).

We can discuss writing the test on the new PR.